In [69]:
library(glue)

# Práctica 4

## Ejercicio 1

Sea $(X, Y)$ un vector aleatorio con función de densidad conjunta:

$
f_{XY}(x,y) =
\begin{cases}
    k(x^2 + y^2) & 2 \leq x \leq 3, 2 \leq y \leq 3 \\
    0 & \text{en otro caso}
\end{cases}
$

### Pregunta A

¿Cuál es el valor de $k$?

$
\begin{align*}
\int_2^3 \int_2^3 k(x^2 + y^2) dx dy
&= k \Big( \int_2^3 x^2 y^2 dx dy \Big) \\
&= k \int_2^{3} \Big[ \tfrac{x^3}{3} + xy^2 \Big]_2^{3} dy \\
&= k \int_2^{3} \tfrac{3^3}{3} + 3y^2 - \tfrac{2^3}{3} - 2y^2 dy \\
&= k \int_2^{3} y^2 + \tfrac{19}{3} dy \\
&= k \Big[ \tfrac{y^3}{3} + \tfrac{19}{3}y \Big]_2^{3} \\
&= k \Big( \tfrac{3^3}{3} + \tfrac{19}{3}3 - \tfrac{2^3}{3} - \tfrac{19}{3}2 \Big) \\
&= k \tfrac{38}{3}
\end{align*}
$

$\iint_{\mathbb{R}^2} f_{XY}(x,y) dx dy = \int_2^3 \int_2^3 k(x^2 + y^2) dx dy = 1 \iff k \tfrac{38}{3} = 1 \iff k = \tfrac{3}{38}$

### Pregunta B

¿Cuál es la probabilidad de que tanto $X$ como $Y$ sean menores que 2.6?

$
\begin{align*}
P(X \leq 2.6, Y \leq 2.6)
&= \int_2^{2.6} \int_2^{2.6} \tfrac{3}{38}(x^2 + y^2) dx dy \\
&= \tfrac{3}{38} \int_2^{2.6} \int_2^{2.6} x^2 + y^2 dx dy \\
&= \tfrac{3}{38} \int_2^{2.6} \Big[ \tfrac{x^3}{3} + xy^2 \Big]_2^{2.6} dy \\
&= \tfrac{3}{38} \int_2^{2.6} \tfrac{2.6^3}{3} + 2.6y^2 - \tfrac{2^3}{3} - 2y^2 dy \\
&= \tfrac{3}{38} \int_2^{2.6} 0.6y^2 + \tfrac{399}{125} dy \\
&= \tfrac{3}{38} \Big[ \tfrac{0.6y^3}{3} + \tfrac{399}{125}y \Big]_2^{2.6} \\
&= \tfrac{3}{38} \Big( \tfrac{0.6*2.6^3}{3} + \tfrac{399}{125}2.6 - \tfrac{0.6*2^3}{3} - \tfrac{399}{125}2 \Big) \\
&= 0.3024
\end{align*}
$

### Pregunta C

¿Cuál es la probabilidad de que $max(X,Y) < 2.6$?

### Pregunta D

Hallar $f_X$ y $f_Y$, las funciones de densidad marginales.

$
\begin{align*}
f_X(x)
&= \int_2^3 \tfrac{3}{38}(x^2 + y^2) dy \\
&= \tfrac{3}{38} \int_2^3 x^2 + y^2 dy \\
&= \tfrac{3}{38} \Big[ yx^2 + \tfrac{y^3}{3} \Big]_2^3 \\
&= \tfrac{3}{38} \Big( 3x^2 + \tfrac{3^3}{3} - 2x^2 - \tfrac{2^3}{3} \Big) \\
&= \tfrac{3}{38} \Big( x^2 + \tfrac{19}{3} \Big) \\
&= \tfrac{3}{38} x^2 + \tfrac{1}{2} \\
\end{align*}
$

Análogamente, $f_Y(y) = \tfrac{3}{38} y^2 + \tfrac{1}{2}$

## Ejercicio 2

De un grupo de tres profesores, dos graduados y un alumno debe seleccionarse al azar una comisión de dos personas. Sean $X$ el número de profesores e $Y$ el número de graduados en la comisión.

### Pregunta A

Hallar la función de probabilidad conjunta del par $(X, Y)$ y las marginales de $X$ e $Y$.

$p_{XY}(1,0) = \frac{{3 \choose 1}{2 \choose 0}{1 \choose 1}}{{6 \choose 2}} = \frac{3}{15}$

$p_{XY}(2,0) = \frac{{3 \choose 2}{2 \choose 0}{1 \choose 0}}{{6 \choose 2}} = \frac{3}{15}$

$p_{XY}(0,1) = \frac{{3 \choose 0}{2 \choose 1}{1 \choose 1}}{{6 \choose 2}} = \frac{2}{15}$

$p_{XY}(1,1) = \frac{{3 \choose 1}{2 \choose 1}{1 \choose 0}}{{6 \choose 2}} = \frac{6}{15}$

$p_{XY}(0,2) = \frac{{3 \choose 0}{2 \choose 2}{1 \choose 0}}{{6 \choose 2}} = \frac{1}{15}$

| Y/X | 0 | 1 | 2 | Py(y) |
|---|---|---|---|---|
| **0** | 0 | 3/15 | 3/15 | 6/15 |
| **1** | 2/15 | 6/15 | 0 | 8/15 |
| **2** | 1/15 | 0 | 0 | 1/15 |
| **Px(x)** | 3/15 | 9/15 | 3/15 | 1 |

### Pregunta B

¿Cuál es la probabilidad de que el alumno no forme parte de la comisión?

$P(X+Y=2) = p_{XY}(0,2) + p_{XY}(2,0) + p_{XY}(1,1) = \tfrac{10}{15} = \tfrac{2}{3}$

## Ejercicio 3

Sea $(X, Y)$ una v.a. bidimensional continua con distribución uniforme en el trapecio de vértices $(-1,0)$, $(0,1)$, $(1,1)$ y $(2,0)$.

### Pregunta A

Hallar la función de densidad conjunta de $(X, Y)$.

Sea $A$ = trapecio de vértices $(-1,0)$, $(0,1)$, $(1,1)$ y $(2,0)$

$
(X,Y) \sim U(A) \iff f_{XY}(x,y) =
\begin{cases}
    k & (x,y) \in A \\
    0 & (x,y) \notin A
\end{cases}
$

Integramos sobre $A$ para buscar el valor de la constante $k$.

$
\int_A f_{XY}(x,y) dx dy = 1 \iff \int_A k dx dy = 1 \iff k \int_A dx dy = 1 \iff k * area(A) = 1 \iff k = \frac{1}{area(A)}
$

$area(A) = \tfrac{1}{2} + 1 + \tfrac{1}{2} = 2 \implies k = \tfrac{1}{2}$

Por lo tanto resulta:

$f_{XY}(x,y) =
\begin{cases}
    \tfrac{1}{2} & (x,y) \in A \\
    0 & (x,y) \notin A
\end{cases}$

### Pregunta B

Calcular $P(Y \leq X)$.

$Y \leq X$ forma un triángulo de vértices $(0,0)$, $(1,1)$, y $(2,0)$.

Llamemos $B$ a este triángulo y notemos que $B \subset A$, y que $area(B) = 1$.

$P(Y \leq X) = \int_B f_{XY}(x,y) dx dy = \int_B \tfrac{1}{2} dx dy = \tfrac{1}{2} * area(B) = \tfrac{1}{2}$

### Pregunta C

Hallar las funciones de densidad marginales $f_X$ y $f_Y$.

Para esto hay que describir al trapecio como una relación entre $x$ e $y$ para poder plantear las integrales.

Podemos considerar: $0 \leq y \leq 1 \land y-1 \leq x \leq 2-y$

Para calcular $f_X$ tenemos que despejar $y$ en función de $x$ para poder plantear la integral. Entonces dividimos el trapecio en 3 partes: el triángulo de la izquierda, el cuadrado del medio, y el triángulo de la derecha.

$
f_X(x)
= \begin{cases}
\int_0^{x+1} \tfrac{1}{2} dy & -1 \leq x < 0 \\
\int_0^1 \tfrac{1}{2} dy & 0 \leq x \leq 1 \\
\int_0^{2-x} \tfrac{1}{2} dy & 1 < x \leq 2
\end{cases}
= \begin{cases}
\tfrac{1}{2} (x+1) & -1 \leq x < 0 \\
\tfrac{1}{2} & 0 \leq x \leq 1 \\
\tfrac{1}{2} (2-x) & 1 < x \leq 2
\end{cases}
$

Otra forma de escribirlo:

$
\begin{align*}
f_X(x)
&= \int_0^{x+1} \tfrac{1}{2} dy + \int_0^1 \tfrac{1}{2} dy + \int_0^{2-x} \tfrac{1}{2} dy \\
&= \tfrac{1}{2} \Big[ y \Big]_0^{x+1} + \tfrac{1}{2} \Big[ y \Big]_0^1 + \tfrac{1}{2} \Big[ y \Big]_0^{2-x} \\
&= \tfrac{1}{2} (x+1) I_{[-1,0)}(x) + \tfrac{1}{2} I_{[0,1]}(x) + \tfrac{1}{2} (2-x) I_{(1,2]}(x)
\end{align*}
$

Para calcular $f_Y$ es más fácil porque solo necesitamos una integral.

$f_Y(y) = \int_{y-1}^{2-y} \tfrac{1}{2} dx = \tfrac{1}{2} \Big[ x \Big]_{y-1}^{2-y} = (\tfrac{3}{2} - y) I_{[0, 1]}(y)$

## Ejercicio 4

Para iluminar sin interrupción una sala (de computadoras!) se cuenta con dos lamparitas; cuando se quema una, se coloca la otra. Sean $X$ e $Y$ los tiempos de vida de cada lamparita (en $10^3$ horas) y supóngase que esos tiempos son independientes y tienen distribución $\mathcal{E}(1)$.

### Pregunta A

Hallar la densidad conjunta de $(X, Y)$.

$f_X = e^{-x} I_{(0, +\infty)}(x)$

$f_Y = e^{-y} I_{(0, +\infty)}(y)$

$f_{XY} = e^{-x} I_{(0, +\infty)}(x) e^{-y} I_{(0, +\infty)}(y) = e^{-(x+y)} I_{(0, +\infty)}(x) I_{(0, +\infty)}(y)$

### Pregunta B

Hallar la probabilidad de que la sala permanezca iluminada al menos 2000 horas.

Como el tiempo de vida de las lamparitas se mide en $10^3$ horas, usamos $\tfrac{2000}{10^3} = 2$ para buscar la probabilidad.

$P(X+Y \geq 2) = 1 - P(X+Y \leq 2) = 1 - \int_0^2 \int_0^{2-x} e^{-(x+y)} dy dx = \tfrac{3}{e^2} \approx 0.406$

Observar que $X+Y \leq 2$ describe un triángulo de vértices $(0,0)$, $(0,2)$ y $(2,0)$.

## Ejercicio 5

Dos servidores A y B procesan trabajos a medida que van llegando. El tiempo que tarda el servidor A en procesar un trabajo es una variable aleatoria $X \sim \mathcal{E}(\lambda_1)$, mientras que el tiempo que tarda el servidor B es una variable aleatoria $Y \sim \mathcal{E}(\lambda_2)$. Ambos servidores actuán en forma independiente. Dos trabajos llegan simultáneamente y son atendidos uno por A y otro B. ¿Cuál es la probabilidad de que el servidor A termine con su trabajo antes que B?

$f_{XY} = \lambda_1 e^{- \lambda_1 x} I_{(0,+\infty)}(x) \lambda_2 e^{- \lambda_2 y} I_{(0,+\infty)}(y)$

$
\begin{align*}
P(X < Y)
&= \int_0^{+\infty} \int_x^{+\infty} \lambda_1 e^{- \lambda_1 x} \lambda_2 e^{- \lambda_2 y} dy dx \\
&= \int_0^{+\infty} \lambda_1 e^{- \lambda_1 x} \Big( \int_x^{+\infty} \lambda_2 e^{- \lambda_2 y} dy \Big) dx \\
&= \int_0^{+\infty} \lambda_1 e^{- \lambda_1 x} \Big[ -e^{- \lambda_2 y} \Big]_x^{+\infty} dx \\
&= \int_0^{+\infty} \lambda_1 e^{- \lambda_1 x} e^{- \lambda_2 x} dx \\
&= \int_0^{+\infty} \lambda_1 e^{-x (\lambda_1 + \lambda_2)} dx \\
&= \tfrac{\lambda_1}{\lambda_1 + \lambda_2} \Big[ -e^{-x (\lambda_1 + \lambda_2)} \Big]_0^{+\infty} \\
&= \tfrac{\lambda_1}{\lambda_1 + \lambda_2}
\end{align*}
$

## Ejercicio 6

Retomemos el problema anterior y supongamos que tres trabajos llegan simultáneamente. Uno es atendido por A, otro por B y el tercero queda esperando a que uno de los servidores se libere. Si $\lambda_1 = \lambda_2$, pruebe que la probabilidad de que el último trabajo en ser atendido sea el último en ser completado es $0.5$.

Sean $X$, $Y$, $Z$ v.a. con distribución $\mathcal{E}(\lambda)$ que miden el tiempo que se tarda en procesar cada uno de las 3 trabajos. Es irrelevante en qué servidor se procesa el trabajo, ya que el parámetro $\lambda$ de la distribución es siempre el mismo. La única condición que tenemos que cuidar es que los servidores procesan solo 1 trabajo a la vez. Si los trabajos $X$ e $Y$ se procesan en un mismo servidor, el tiempo total transcurrido en ese servidor sería $X+Y$.

Tenemos 2 posibles escenarios tales que el tercer trabajo es el último en terminar:
- El trabajo $X$ tarda menos que $Y$, y por lo tanto es el servidor A quien procesa el trabajo $Z$. En este caso, necesitamos calcular la probabilidad de que $X+Z>Y$ para afirmar que el tercer trabajo fue el último en terminar.
- De forma análoga, si el trabajo $X$ tarda más que $Y$, entonces es el servidor B quien procesa el trabajo $Z$. En este caso, necesitamos calcular la probabilidad de que $Y+Z>X$.

Sigue un diagrama que intenta explicar estos 2 escenarios:

```
A --X--|--Z--        A ---X---
B ---Y---            B --Y--|--Z--
```

Recordemos que por el ejercicio anterior concluimos que:

$P(X<Y) = \tfrac{\lambda_1}{\lambda_1 + \lambda_2}$

Como $\lambda_1 = \lambda_2 \implies P(X<Y) = \tfrac{\lambda_1}{2 \lambda_1} = \tfrac{1}{2}$

Y por el complemento vale: $P(X>Y) = 1 - P(X<Y) = \tfrac{1}{2}$

Ahora sí, hagamos las cuentas. Primero calculamos la probabilidad del primer escenario.

$\begin{align*}
P(X+Z>Y, Y>X)
&= P(X+Z>Y|Y>X)P(Y>X) \\
&= (1 - P(Y>X+Z|Y>X))P(Y>X) \text{ Calculamos el complemento para dar vuelta la desigualdad} \\
&= (1 - P(Y>Z))P(Y>X) \text{ Usamos la propiedad de falta de memoria} \\
&= (1 - \tfrac{1}{2})\tfrac{1}{2} \text{ Usamos el resultado del ejercicio anterior} \\
&= \tfrac{1}{4}
\end{align*}
$

De forma análoga, el segundo escenario tiene la misma probabilidad:

$\begin{align*}
P(Y+Z>X, X>Y)
&= P(Y+Z>X|X>Y)P(X>Y) \\
&= (1 - P(X>Y+Z|X>Y))P(X>Y) \\
&= (1 - P(X>Z))P(X>Y) \\
&= (1 - \tfrac{1}{2})\tfrac{1}{2} \\
&= \tfrac{1}{4}
\end{align*}
$

Finalmente, la probabilidad de que el tercer trabajo sea el último en completarse es:

$P(X+Z>Y, Y>X) + P(Y+Z>X, X>Y) = \tfrac{1}{4} + \tfrac{1}{4} = \tfrac{1}{2}$

## Ejercicio 7

En los ejercicios 1 a 3:

### Pregunta A

Decir si $X$ e $Y$ son independientes (justificando en cada caso).

**Ejercicio 1**

$X$ e $Y$ son independientes $\iff f_{XY}(x,y) = f_X(x)f_Y(y)$

$f_{XY}(x,y) = \tfrac{3}{38}(x^2 + y^2) \neq (\tfrac{3}{38} x^2 + \tfrac{1}{2})(\tfrac{3}{38} y^2 + \tfrac{1}{2}) = f_X(x) f_Y(y)$

Además se puede probar que no son independientes mostrando un punto en donde no se satisface la igualdad. Esto vale porque todas las densidades son funciones continuas en un entorno del $(2,2)$.

$f_{XY}(2,2) = \tfrac{12}{19} \neq \tfrac{961}{1444} = f_X(2) f_Y(2)$

$\implies$ $X$ e $Y$ **no** son independientes

**Ejercicio 2**

$p_{XY}(0,2) = \tfrac{1}{15} \neq \tfrac{3}{15} \tfrac{1}{15} = p_X(0) p_Y(2)$

$\implies$ $X$ e $Y$ **no** son independientes

**Ejercicio 3**

Como la región es un trapecio, al multiplicar $f_Xf_y$ las funciones indicadores terminan describiendo un rectángulo, incluyendo puntos que no están dentro del trapecio.

$\implies$ $X$ e $Y$ **no** son independientes

### Pregunta B

Hallar las funciones de probabilidad puntual condicional o de densidad condicional, según corresponda.

**Ejercicio 1**

$f_{X|Y=y}(x) = \tfrac{f_{XY}(x,y)}{f_Y(y)} = \tfrac{\tfrac{3}{38} (x^2 + y^2) I_{[2,3]}(x) I_{[2,3]}(y)}{(\tfrac{3}{38} y^2 + \tfrac{1}{2})I_{[2,3]}(y)}$

$f_{Y|X=x}(y) = \tfrac{f_{XY}(x,y)}{f_X(x)} = \tfrac{\tfrac{3}{38} (x^2 + y^2) I_{[2,3]}(x) I_{[2,3]}(y)}{(\tfrac{3}{38} x^2 + \tfrac{1}{2})I_{[2,3]}(x)}$

**Ejercicio 2**

**Ejercicio 3**

## Ejercicio 8

Se tira una moneda equilibrada 3 veces, siendo $X$ el número de caras. Si $X = a$ se extraen sin reposición $a + 1$ bolillas de una urna que contiene 4 bolillas blancas y una roja. Sea $Y$ el número de bolillas rojas extraídas.

$X \sim Bi(3, 0.5)$

$Y \sim H(x+1,5,1)$

### Pregunta A

Hallar la distribución de $Y$ dado $X = a$, para $a = 0,1,2,3$.

$P(Y=y|X=a) = \tfrac{{1 \choose y}{4 \choose a+1-y}}{5 \choose a+1}$

In [102]:
p = function(y, a) choose(1,y) * choose(4, a+1-y) / choose(5, a+1)
Rx = 0:3
Ry = 0:1
t = rbind(p(0, Rx), p(1, Rx))
rownames(t) = c(glue('Y={Ry}'))
colnames(t) = c(glue('P(Y|X={Rx})'))
t

P(Y|X=0) P(Y|X=1) P(Y|X=2) P(Y|X=3)
Y=0 0.8      0.6      0.4      0.2     
Y=1 0.2      0.4      0.6      0.8

### Pregunta B

Obtener una tabla con la distribución conjunta del par $(X, Y)$ y hallar la función de probabilidad puntual marginal $p_Y$.

$P(X=x, Y=y) = P(X=x \cap Y=y) = P(Y=y|X=x)P(X=x) = \tfrac{{1 \choose y}{4 \choose x+1-y}}{5 \choose x+1} {3 \choose x} 0.5^x 0.5^{3-x}$

In [101]:
p = function(x, y) dhyper(y, 1, 4, x+1) * dbinom(x, 3, 0.5)
Rx = 0:3
Ry = 0:1
p0 = p(Rx, 0)
p1 = p(Rx, 1)
t = rbind(c(p0, sum(p0)), c(p1, sum(p1)))
rownames(t) = c(glue('Y={Ry}'))
colnames(t) = c(glue('X={Rx}'), 'pY')
t

X=0   X=1   X=2   X=3   pY 
Y=0 0.100 0.225 0.150 0.025 0.5
Y=1 0.025 0.150 0.225 0.100 0.5

### Pregunta C

¿Son $X$ e $Y$ independientes?

Intuitivamente podemos decir que no son independientes, ya que la cantidad de bolillas que se van a extraer dependen de la cantidad de caras que hayan salido con la moneda.

Formalmente podemos justificarlo con un contraejemplo:

$X$ e $Y$ son independientes $\iff p_{XY}(x,y) = p_X(x)p_Y(y)$

$p_{XY}(0,0) = 0.1$

$p_X(0)p_Y(0) = 0.125 * 0.5 = 0.0625 \neq 0.1$

### Pregunta D

Si se extrajeron 2 bolillas blancas, ¿cuál es la probabilidad de que hayan salido dos caras?

Sea $Z$ = cantidad de bolillas blancas

$Z \sim H(x+1,5,4)$

$P(X=2|Z=2) = \tfrac{P(Z=2|X=2)P(X=2)}{P(Z=2)}$

Armemos otra tabla de probabilidad conjunta $(X, Z)$ para poder encontrar luego $P(Z=2)$.

In [103]:
p = function(x, z) dhyper(z, 4, 1, x+1) * dbinom(x, 3, 0.5)
Rx = 0:3
Rz = 0:4
p0 = p(Rx, 0)
p1 = p(Rx, 1)
p2 = p(Rx, 2)
p3 = p(Rx, 3)
p4 = p(Rx, 4)
t = rbind(c(p0, sum(p0)), c(p1, sum(p1)), c(p2, sum(p2)), c(p3, sum(p3)), c(p4, sum(p4)))
rownames(t) = c(glue('Z={Rz}'))
colnames(t) = c(glue('X={Rx}'), 'pZ')
t

X=0   X=1   X=2   X=3   pZ   
Z=0 0.025 0.000 0.000 0.000 0.025
Z=1 0.100 0.150 0.000 0.000 0.250
Z=2 0.000 0.225 0.225 0.000 0.450
Z=3 0.000 0.000 0.150 0.100 0.250
Z=4 0.000 0.000 0.000 0.025 0.025

Por lo tanto:

$P(X=2|Z=2) = \tfrac{P(Z=2|X=2)P(X=2)}{P(Z=2)} = \tfrac{0.6 * 0.375}{0.45} = 0.5$

**Otra forma** (más práctica y con menos cuentas) de calcular esto sería mirar la tabla de probabilidad conjunta $(X,Y)$ y calcular la probabilidad de que hayan salido dos caras y 2 bolillas blancas dividido por la probabilidad de que hayan salidos 2 bolillas blancas.

Probabilidad de que hayan salido dos caras y 2 bolillas blancas $= P(X=2,Y=1) = 0.225$

Probabilidad de que hayan salidos 2 bolillas blancas $= P(X=1,Y=0) + P(X=2,Y=1) = 0.225 + 0.225 = 0.45$

$\implies \tfrac{0.225}{0.45} = 0.5$